File: Week_4/OOPS/InheritanceDemo.java
class Box {
    protected double length;
    protected double breadth;

    public Box(double length, double breadth) {
        this.length = length;
        this.breadth = breadth;
    }

    public double findArea() {
        return length * breadth;
    }
}

class Box3d extends Box {
    private double height;

    public Box3d(double length, double breadth, double height) {
        super(length, breadth);
        this.height = height;
    }

    public double findVolume() {
        return findArea() * height;
    }
}

public class InheritanceDemo {
    public static void main(String[] args) {
        Box3d myBox = new Box3d(5.0, 4.0, 3.0);
        
        System.out.println("Surface Base Area: " + myBox.findArea());
        System.out.println("Total Volume: " + myBox.findVolume());
    }
}

File: Week_4/OOPS/incpp/InheritanceDemo.cpp
#include <iostream>

class Box {
protected:
    double length;
    double breadth;

public:
    Box(double length, double breadth) {
        this->length = length;
        this->breadth = breadth;
    }

    double findArea() {
        return length * breadth;
    }
};

class Box3d : public Box {
private:
    double height;

public:
    Box3d(double length, double breadth, double height) : Box(length, breadth) {
        this->height = height;
    }

    double findVolume() {
        return findArea() * height;
    }
};

int main() {
    Box3d myBox(5.0, 4.0, 3.0);

    std::cout << "Surface Base Area: " << myBox.findArea() << std::endl;
    std::cout << "Total Volume: " << myBox.findVolume() << std::endl;

    return 0;
}
